### PR TITLE
[fix](decommission) fix cann't decommission mtmv

### PR DESCRIPTION
**Pull request by yujun777**
BUG:  

For function Table.needSchedule() , it only check with table type equals OLAP.  But MTMV's  table type is MATERIALIZED_VIEW,  so MTMV cann't  be repair and decommission. To fix this,  Table.needSchedule should check with table types = OLAP + MATERIALIZED_VIEW.  And Table.isManagedTable(include OLAP + MATERIALIZED_VIEW) can achieve this goal.

But considering Table.needSchedule is easy to misuse. It's not exclude colocate table,  we delete this function.
 

What's more, this PR also fix:
1. colocate rebalance exclude host should consider Config.allow_replica_on_same_host;

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java [hunk 1]
```diff
@@ -314,7 +314,7 @@ private static void checkDecommissionWithReplicaAllocation(List<Backend> decommi
             for (Table table : db.getTables()) {
                 table.readLock();
                 try {
-                    if (!table.needSchedule()) {
+                    if (!table.isManagedTable()) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java:317]: Is this the same semantic as the original impl.?

Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java:317]: the original has bug,  should check MTMV here too.

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run p0

**Comment by dataroaring** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zddr — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**